Makd.mak

```diff
@@ -565,6 +565,40 @@ $O/test-%.stamp: $O/test-%
 	$(call exec,$< $(ITFLAGS),$<,run)
 	$Vtouch $@
 
+# examples rules
+##########################
+
+# examples are assumed to be standalone programs, so we recursively search
+# for files examples/*.d and assume that each file is a separate example
+# program. Files can exist in sub-directories under examples/ dir.
+# The sources list is filtered through the $(EXAMPLES_FILTER_OUT) variable 
+# contents (using the Make function $(filter-out)), so you can exclude an
+# example by adding the example file path (as an absolute path using $C) to
+# this variable.
+
+# The 'example' target builds all the examples.
+.PHONY: example
+example: $(patsubst $T/example/%.d,$O/example/%,\
+	$(filter-out $(EXAMPLE_FILTER_OUT),$(shell find $T/example/ -name "*.d")))
+
+# General rule to build an example program.
+$O/example/%: BUILD.d.depfile = $O/example/$*.mak
+$O/example/%: $T/example/%.d $G/build-d-flags
+	$(call build_d)
```

Comment:
I guess is probably a good idea to compile examples with unittests, otherwise unittests will never be run. And BTW, I wouldn't run examples unittests as part of the regular tests... (see #41).